client/result/result.go

```diff
@@ -0,0 +1,352 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
```

Comment:
The `Result` object is not in the `client` package, it's in `client/result`.

My rationale for this package structure: originally, the `client` package depended upon `bootstrap` because of `bootstrap.Result` object. That felt like a more brittle dependency. So instead, I introduced a new package for the `client` specific parts of the `Result` object. This package didn't feel heavy enough to merit a new top-level package, so I stuck it in `client/result`. This way, `client` and `bootstrap` both depend on `client/Result`.

Am I overthinking this?